public/app/features/explore/state/utils.test.ts

```diff
@@ -30,17 +35,51 @@ describe('loadAndInitDatasource', () => {
     expect(dataSourceMock.get).toBeCalledWith({ uid: 'Unknown' });
     expect(dataSourceMock.get).toBeCalledWith();
     expect(instance).toMatchObject(DEFAULT_DATASOURCE);
-    expect(store.set).toBeCalledWith(lastUsedDatasourceKeyForOrgId(1), DEFAULT_DATASOURCE.uid);
+    expect(setLastUsedDatasourceUIDSpy).toBeCalledWith(1, DEFAULT_DATASOURCE.uid);
   });
 
   it('saves last loaded data source uid', async () => {
+    setLastUsedDatasourceUIDSpy = jest.spyOn(exploreUtils, 'setLastUsedDatasourceUID');
     dataSourceMock.get.mockResolvedValue(TEST_DATASOURCE);
 
     const { instance } = await loadAndInitDatasource(1, { uid: 'Test' });
 
     expect(dataSourceMock.get).toBeCalledTimes(1);
     expect(dataSourceMock.get).toBeCalledWith({ uid: 'Test' });
     expect(instance).toMatchObject(TEST_DATASOURCE);
-    expect(store.set).toBeCalledWith(lastUsedDatasourceKeyForOrgId(1), TEST_DATASOURCE.uid);
+    expect(setLastUsedDatasourceUIDSpy).toBeCalledWith(1, TEST_DATASOURCE.uid);
+  });
+});
+
+describe('getRange', () => {
+  it('should parse moment date', () => {
+    // convert date strings to moment object
+    const range = { from: dateTime('2020-10-22T10:44:33.615Z'), to: dateTime('2020-10-22T10:49:33.615Z') };
+    const result = getRange(range, 'browser');
+    expect(result.raw).toEqual(range);
+  });
+
+  it('should parse epoch strings', () => {
+    const range = {
+      from: dateTime('2020-10-22T10:00:00Z').valueOf().toString(),
+      to: dateTime('2020-10-22T11:00:00Z').valueOf().toString(),
+    };
+    const result = getRange(range, 'browser');
```

Comment:
Assertions in both tests are the same, they could be extracted to a seperate function.

Reply:
i think separating assertions in tests from the actual test body it's a bad practice most of the times as it makes both debugging tests harder and errors harder to follow should the test itself stop working for some reason.